### PR TITLE
chore(windows): split keyman64 header to keymanengine

### PR DESCRIPTION
**Pull request by rc-swag**
Ground work for #5011 . This PR breaks out the include files and definitions from keyman64.h into keymanengine.h This will allow for projects like kmanalyze to just include keyman64.h and not keymanengine.h and therefore the whole keyman core library.

### CODE REVIEW

File: windows/src/global/inc/keymanengine.h [hunk 1]
```diff
@@ -0,0 +1,254 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * Name:             KeymanEngine
+ * Description:      Main header file for windows engine
+ *
+*/
+/***************************************************************************/   // I4006   // I4169
```
Comment (rc-swag) [inline windows/src/global/inc/keymanengine.h:8]: Not sure how this version history works for e.g. `// I4006` but I have brought them all over throughout this file.

File: windows/src/global/inc/keymanengine.h [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+DWORD EthnologueStringCodeToDWord(PWSTR EthCode);
+
+PWSTR  GetSystemStore(LPKEYBOARD kb, DWORD SystemID);
+
+DWORD ExceptionMessage(LPSTR Proc, LPEXCEPTION_POINTERS ep);
+
+void keybd_shift(LPINPUT pInputs, int* n, BOOL isReset, LPBYTE const kbd);
+
+//#define KEYEVENT_EXTRAINFO_KEYMAN 0xF00F0000   // I4370
+#ifndef _KEYMAN64_LIGHT
```
Comment (rc-swag) [inline windows/src/global/inc/keymanengine.h:217]: Should we update this define (and therefore all the places it is defined in projects) to KEYMANENGINE_LIGHT?

File: windows/src/global/inc/keymanengine.h [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+#include "k32_tsf.h"
+
+void ReportActiveKeyboard(WORD wCommand);   // I3933   // I3949
+void SelectKeyboardHKL(DWORD hkl, BOOL foreground);  // I3933   // I3949   // I4271
+BOOL SelectKeyboardTSF(DWORD KeymanID, BOOL foreground);   // I3933   // I3949   // I4271
+BOOL ReportKeyboardChanged(WORD wCommand, DWORD dwProfileType, UINT langid, HKL hkl, GUID clsid, GUID guidProfile);
+void ProcessModifierChange(UINT key, BOOL isUp, BOOL isExtended);   // I4793
+
+#endif  // _KEYMAN64_LIGHT
+#endif	// _KEYMANENGINE_H
```
Comment (rc-swag) [inline windows/src/global/inc/keymanengine.h:254]: fix indents

File: windows/src/global/inc/keymanengine.h (hunk 1 above)
Comment (mcdurdin) [reply windows/src/global/inc/keymanengine.h:8]: Yeah, these are legacy issue references. We have the legacy issues in a private repo, e.g. https://github.com/sillsdev/keyman-issues/issues, so you can search for the tag `I4006` which returns https://github.com/sillsdev/keyman-issues/issues/3997.

File: windows/src/global/inc/keymanengine.h (hunk 2 above)
Comment (mcdurdin) [reply windows/src/global/inc/keymanengine.h:217]: In theory we should be able to remove it once we finish all the refactoring as I reckon its use is another example of this patching.

**Review by mcdurdin — APPROVED** [review]
LGTM.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.90-alpha](https://keyman.com/downloads/releases/alpha/15.0.90)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.90-alpha](https://keyman.com/downloads/releases/alpha/15.0.90)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.90-alpha](https://keyman.com/downloads/releases/alpha/15.0.90)

**Pull request merged**